Today Extension/TodayViewController.swift

```diff
@@ -0,0 +1,226 @@
+
+        setUpLocation()
+        setUpRoutesTableView()
+        view.addSubview(routesTable)
+        createConstraints()
+
+        favorites = TodayExtensionManager.shared.retrieveFavoritesNames()
+        coordinates = TodayExtensionManager.shared.retrieveFavoritesCoordinates()
+
+        searchForRoutes()
+    }
+
+    func searchForRoutes() {
+        if
+            favorites.count > 0,
+            let start = currentLocation {
+                Network.getMultiRoutes(startCoord: start, time: Date(), endCoords: coordinates, endPlaceNames: favorites) { (request) in
+                    self.processRequest(request: request)
+                }
+            }
+    }
+
+    func processRequest(request: APIRequest<MultiRoutesRequest, Error>) {
+        request.performCollectingTimeline { (response) in
+            switch response.result {
+            case .success(let routesResponse):
+                self.routes = routesResponse.data
+                self.rearrangeRoutes()
+                self.didFetchRoutes = true
+                self.extensionContext?.widgetLargestAvailableDisplayMode = .expanded
+                self.routesTable.reloadData()
+            case .failure(let networkError):
+                if let error = networkError as? APIError<Error> {
+                    self.processRequestError(error: error)
+                }
+            }
+        }
+    }
+
+    func processRequestError(error: APIError<Error>) {
+        let title = "Network Failure: \((error.error as NSError?)?.domain ?? "No Domain")"
+        let description = (error.localizedDescription) + ", " + ((error.error as NSError?)?.description ?? "n/a")
+
+        routes = []
+        print("Error Title: \(title)")
+        print("Error Description: \(description)")
+    }
+
+    /// Called in response to the user tapping the “Show More” or “Show Less” buttons
+    func widgetActiveDisplayModeDidChange(_ activeDisplayMode: NCWidgetDisplayMode, withMaximumSize maxSize: CGSize) {
+        let expanded = activeDisplayMode == .expanded
+        preferredContentSize = expanded ? CGSize(width: maxSize.width, height: 110.0 * CGFloat(favorites.count)) : maxSize
+
+        routesTable.reloadData()
+    }
+
+    func createConstraints() {
+        routesTable.snp.makeConstraints {(make) in
+            make.leading.trailing.equalToSuperview()
+            make.height.equalTo(110.0*5)
+        }
+    }
+
+    /// Re-orders routes so that walking-only routes are at the end of the list of routes. Modifies the list of favorites and coordinates coorespondingly.
+    func rearrangeRoutes() {
+        var nonNilRoutes = [Route?]()
+        var nonNilFavorites = [String]()
+        var nonNilCoordinates = [String]()
+
+        var nilRoutes = [Route?]()
+        var nilFavorites = [String]()
+        var nilCoordinates = [String]()
+
+        for i in 0..<routes.count {
+
+            if (routes[i]?.directions.filter { $0.type == .depart })?.first != nil {
```

Comment:
`.sort` by `$0.type == .depart` to reduce all the variable usage

Reply:
Changed up the logic a little bit for this function, but don't think it's possible to just use `.sort`. I need to sort the `favorites` array and `coordinates` array according to how `routes` is re-arranged, and I need to keep track of the index for that.